x/programs/rust/wasmlanche-sdk/src/memory.rs

```diff
@@ -114,20 +114,4 @@ mod tests {
         // see https://doc.rust-lang.org/1.77.2/std/alloc/struct.Layout.html#method.array
         alloc(isize::MAX as usize + 1);
     }
-
-    // TODO these two tests make the code abort and not panic, it's hard to write an assertion here
-    // #[test]
-    // #[should_panic]
-    // fn two_big_allocation_fails() {
-    //     // see https://doc.rust-lang.org/1.77.2/std/alloc/struct.Layout.html#method.array
-    //     alloc((isize::MAX / 2) as usize + 1);
-    //     alloc((isize::MAX / 2) as usize + 1);
-    // }
-
-    //     #[test]
-    //     #[should_panic]
-    //     fn null_pointer_allocation() {
-    //         // see https://doc.rust-lang.org/1.77.2/std/alloc/struct.Layout.html#method.array
-    //         alloc(isize::MAX as usize);
-    //     }
```

Comment:
We might have a chance running these in a thread, in which case I will remove that change and write tests in another PR. otherwise will leave as-is.

Reply:
Given that "abort" means that the program won't do anything else, it seems unlikely to be able to catch these signals. Running a function in a new process is not really a thing either. see https://stackoverflow.com/questions/58242918/is-it-possible-to-run-a-function-as-a-process-in-rust